pkg/controllers/v1alpha1/datatable/datatable_controller.go

```diff
@@ -83,18 +89,23 @@ func (r *DataTableReconciler) Reconcile(context context.Context, req ctrl.Reques
 
 	if err := r.Get(ctx, req.NamespacedName, &ctx.Datatable); err != nil {
 		// can not get the datatable obj
-		// TODO: err handle
-		ctx.Log.Error(err, "Fail to get the datatable", req)
-		return utils.RequeueIfError(err)
+		if utils.IgnoreNotFound(err) != nil {
+			ctx.Log.Error(err, "Fail to get the datatable", req)
+			return utils.RequeueIfError(err)
+		} else {
+			ctx.Log.V(1).Info("Fail to get the datatable")
+			return ctrl.Result{}, nil
+		}
 	} else {
 		return r.reconcileDataTable(ctx)
 	}
 }
 
 func (r *DataTableReconciler) reconcileDataTable(ctx reconcileRequestContext) (result ctrl.Result, err error) {
 
-	ip, err := datatable.GetOrCreateAlluxio(r.Client, CommonName, ctx.Datatable.Namespace)
-	if err != nil || len(ip) == 0 {
+	// 0. create dataset and alluxio runtime if they do not exist, or get the master ip
+	MasterIP, err = datatable.GetOrCreateAlluxio(r.Client, CommonName, ctx.Datatable.Namespace)
```

Comment:
why not keeping masterIp in datatable's status?

Reply:
fixed